client/src/pages/reports/Form.js

```diff
@@ -550,28 +559,31 @@ export default class ReportForm extends ValidatableFormWrapper {
 
 	@autobind
 	autoSave() {
-		let timeoutId = window.setTimeout(this.autoSave, 30000)
-		this.setState({timeoutId})
-
-		//If the report hasn't changed, don't save it.
-		if (this.state.reportChanged === false) { return }
-
-		this.saveReport(false)
-			.then(response => {
-				if (response.id) {
-					this.props.report.id = response.id
-				}
-				if (response.reportSensitiveInformation) {
-					this.props.report.reportSensitiveInformation = response.reportSensitiveInformation
-				}
+		// Only auto-save if the report has changed
+		if (this.state.reportChanged === false) {
+			this.autoSaveTimeout = this.defaultTimeout.clone() // reset to default
+		}
+		else {
+			this.saveReport(false)
+				.then(response => {
+					if (response.id) {
+						this.props.report.id = response.id
+					}
+					if (response.reportSensitiveInformation) {
+						this.props.report.reportSensitiveInformation = response.reportSensitiveInformation
+					}
 
-				//Reset the reportchanged state, yes this could drop a few keystrokes that
-				// the user made while we were saving, but that's not a huge deal.
-				this.setState({autoSavedAt: moment(), reportChanged: false, showAutoSaveBanner: true})
-				window.setTimeout(this.hideAutoSaveBanner, 5000)
-			}).catch(response =>
-				this.setState({error: "There was an error autosaving your report. We'll try again in a few seconds"})
-			)
+					//Reset the reportChanged state, yes this could drop a few keystrokes that
+					// the user made while we were saving, but that's not a huge deal.
+					this.setState({autoSavedAt: moment(), reportChanged: false, showAutoSaveBanner: true, autoSaveError: null})
```

Comment:
shouldn't we reset back to the default timeout sometime around here?

Reply:
Indeed.